### PR TITLE
feat: Add accessibility string to OtherUserProfile and search participants [WPB-9784]

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9784" title="WPB-9784" target="_blank">TaskWPB-9784</a>  [Android] accessibility strings – Conversation details
  </td></table>
  <br />
 

# What's new in this PR?

Second part of PR https://github.com/wireapp/wire-android/pull/3495
Was split just to make review easier.

Added accessibility strings to OtherUserProfile and SearchParticipants screens.

### CODE REVIEW

**Review by Garzas — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/common/CopyButton.kt [hunk 1]
```diff
@@ -27,13 +28,14 @@ import com.wire.android.ui.common.button.WireSecondaryIconButton
...
-    modifier: Modifier = Modifier
+    modifier: Modifier = Modifier,
+    @StringRes contentDescription: Int? = null,
+    state: WireButtonState = WireButtonState.Default
 ) {
     WireSecondaryIconButton(
         onButtonClicked = onCopyClicked,
         iconResource = R.drawable.ic_copy,
-        contentDescription = R.string.content_description_copy,
+        contentDescription = contentDescription ?: R.string.content_description_copy,
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/common/CopyButton.kt]: i would be in favour of removing this implicit behaviour.with a default value for  @StringRes content description

```kotlin
@Composable
fun CopyButton(
    onCopyClicked: () -> Unit,
    state: WireButtonState = WireButtonState.Default,
    modifier: Modifier = Modifier
    modifier: Modifier = Modifier,
    @StringRes contentDescription: Int = R.string.content_description_copy,
    state: WireButtonState = WireButtonState.Default
) {
    WireSecondaryIconButton(
        onButtonClicked = onCopyClicked,
        iconResource = R.drawable.ic_copy,
        contentDescription = contentDescription,

```

File: app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextFieldLayout.kt [hunk 2]
```diff
@@ -101,8 +109,8 @@ internal fun WireTextFieldLayout(
                 .fillMaxWidth()
                 .background(color = colors.backgroundColor(state).value, shape = shape)
                 .border(width = dimensions().spacing1x, color = colors.borderColor(state, interactionSource).value, shape = shape)
-                .semantics {
-                    (labelText ?: placeholderText ?: descriptionText)?.let {
+                .semantics(mergeDescendants = true) {
+                    (semanticDescription ?: labelText ?: placeholderText ?: descriptionText)?.let {
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextFieldLayout.kt:113]: same as above moving the default values to the function parameters makes it easier to understands what is going on + makes it able to set it to null when there is a need to do so

**Review by MohamadJaara — CHANGES_REQUESTED** [review]
some suggestions about removing implicant behaviours

File: app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextFieldLayout.kt (hunk 2 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextFieldLayout.kt:113]: answered to this in here https://github.com/wireapp/wire-android/pull/3541#discussion_r1812376271

**Review by MohamadJaara — APPROVED** [review]

**Pull request merged**